changes/6861.fixed

```diff
@@ -0,0 +1 @@
+Fixed the data population of the Rack Group dropdown to include Rack Groups from the Rack's parent locations in Rack Edit View.
```

Comment:
Interesting - so was this scenario already permitted through the ORM and REST API? Is the clean logic for Rack/RackGroup assignment already correct, or is validation simply missing altogether for relative locations?

Reply:
It looks like we do have that logic in place for `Rack.clean()` https://github.com/nautobot/nautobot/blob/develop/nautobot/dcim/models/racks.py#L209-L219. Only the data population is incorrect in the dropdown